### PR TITLE
fix: coop prop ping

### LINKED ISSUE #177: Ping tool in coop no longer works when pinging props
Valve broke it recently. Perhaps it can be fixed through the server plugin.

### PR DESCRIPTION
**Pull request by ThisAMJ**
Fixes a base game bug (caused by an update) where players are unable to ping puzzle elements. Linux doesn't work yet, I'm working on it

Fixes #177

### CODE REVIEW

File: src/Modules/Server.cpp [hunk 1]
```diff
@@ -708,6 +709,17 @@ float hostTimeWrap() {
 	return engine->GetHostTime();
 }
 
+// UTIL_EntityByIndex
+extern Hook g_EntityByIndexHook;
+DETOUR_T(int, Server::EntityByIndex, int entityIndex) {
+	if (entityIndex > Offsets::NUM_ENT_ENTRIES) return NULL;
```
Comment (NeKzor) [inline src/Modules/Server.cpp:715]: I guess this is supposed to be the real bugfix that "Valve" missed?

Also `NULL` should only be used for pointers:
```suggestion
	if (entityIndex > Offsets::NUM_ENT_ENTRIES) return 0;
```

File: src/Modules/Server.cpp [hunk 2]
```diff
@@ -810,6 +822,38 @@ bool Server::Init() {
...
+	uintptr_t code_a = (uintptr_t)(PlayerClientCommand) + 458;
+	if (*(uint8_t *)code_a == 0x7F) {
+		Memory::UnProtect((void *)code_a, 1);
+		*(uint8_t *)code_a = 0x74;
+	}
+	uintptr_t code_b = code_a + 32;
+	if (*(uint8_t *)code_b == 0x7F) {
+		Memory::UnProtect((void *)code_b, 1);
+		*(uint8_t *)code_b = 0x74;
+	}
```
Comment (NeKzor) [inline src/Modules/Server.cpp:855]: Can you add a comment on what all of this code here does?
The first offset into the function is unlikely to be the same on Windows and Linux.

**Review by NeKzor — CHANGES_REQUESTED** [review]

File: src/Modules/Server.cpp (hunk 1 above)
Comment (ThisAMJ) [reply src/Modules/Server.cpp:715]: The function we're hooking uses NULL but sure

File: src/Modules/Server.cpp (hunk 2 above)
Comment (ThisAMJ) [reply src/Modules/Server.cpp:855]: This turns conditional jumps (i forget if its gt or lt) for entity index 32 and 33 in the `signify` client command into jz which will never jump afaik

this fixes the ping not working for props, other hook just fixes a potential crash exploit for orange

Comment (ThisAMJ) [reply src/Modules/Server.cpp:855]: have you *tried* it on linux?

File: src/Modules/Server.cpp (hunk 1 above)
Comment (ThisAMJ) [reply src/Modules/Server.cpp:715]: ok so `UTIL_EntityByIndex` returns type `CBaseEntity *`

**Pull request closed without merging**